ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/adoptopenjdk_install/tasks/main.yml

```diff
@@ -115,142 +129,180 @@
+      shell: curl -s 'https://api.adoptium.net/v3/assets/feature_releases/{{ jdk_version }}/ga?architecture={{ api_architecture }}&heap_size=normal&image_type=jdk&jvm_impl={{ bootjdk }}&os={{ platformLinux }}&page=0&page_size=1&project=jdk&vendor=eclipse' | grep signature_link | awk '{split($0,a,"\""); print a[4]}'
+      when:
+        - jdk_version != 10
+        - not (jdk_version == 8 and ansible_architecture == "s390x")
+      register: sig_output
 
-- name: Install latest release if one not already installed (alpine-linux)
-  unarchive:
-    src: https://api.adoptopenjdk.net/v3/binary/latest/{{ jdk_version }}/ga/alpine-linux/{{ api_architecture }}/jdk/{{ bootjdk }}/normal/adoptopenjdk?project=jdk
-    dest: /usr/lib/jvm
-    remote_src: yes
-  retries: 3
-  delay: 5
-  register: adoptopenjdk_download
-  until: adoptopenjdk_download is not failed
-  when:
-    - ansible_distribution == "Alpine"
-    - adoptopenjdk_installed.rc != 0
-  tags: adoptopenjdk_install
+    - name: Download latest release (Linux/Alpine-Linux)
+      get_url:
+        url: https://api.adoptopenjdk.net/v3/binary/latest/{{ jdk_version }}/ga/{{ platformLinux }}/{{ api_architecture }}/jdk/{{ bootjdk }}/normal/adoptopenjdk?project=jdk
+        dest: /tmp/jdk{{ jdk_version }}.tar.gz
+        mode: 0440
+      retries: 3
+      delay: 5
+      register: adoptopenjdk_download
+      until: adoptopenjdk_download is not failed
+
+    - name: GPG Signature verification (Linux/Alpine-Linux)
+      script: ../Supporting_Scripts/package_signature_verification.sh -f /tmp/jdk{{ jdk_version }}.tar.gz -sl "{{ sig_output.stdout }}" -k {{ key.adoptium }}
+      when:
+        - jdk_version != 10
+        - not (jdk_version == 8 and ansible_architecture == "s390x")
 
-# CentOS6 needs it's own task so it can use a different python interpreter.
-# See: https://github.com/adoptium/infrastructure/issues/1877
-- name: Install latest release if not already installed (CentOS6)
-  unarchive:
-    src: https://api.adoptopenjdk.net/v3/binary/latest/{{ jdk_version }}/ga/linux/{{ api_architecture }}/jdk/{{ bootjdk }}/normal/adoptopenjdk?project=jdk
-    dest: /usr/lib/jvm
-    remote_src: yes
-  vars:
-    ansible_python_interpreter: /usr/local/python2/bin/python2.7
-  retries: 3
-  delay: 5
-  register: adoptopenjdk_download
-  until: adoptopenjdk_download is not failed
+    - name: Install latest release if one not already installed (Linux/Alpine-Linux)
+      unarchive:
+        src: /tmp/jdk{{ jdk_version }}.tar.gz
+        dest: /usr/lib/jvm
+        remote_src: yes
+
+    - name: Remove jdk{{ jdk_version }}.tar.gz (Linux/Alpine-Linux)
+      file:
+        path: /tmp/jdk{{ jdk_version }}.tar.gz
+        state: absent
+
+# # CentOS6 needs it's own task so it can use a different python interpreter.
+# # See: https://github.com/adoptium/infrastructure/issues/1877
+- name: Install latest JDK {{ jdk_version }} release if not already installed (CentOS6)
```

Comment:
are we still supporting CentOS 6?

Reply:
So long as we are still building and pushing the centos6 docker image, https://github.com/adoptium/infrastructure/blob/master/ansible/docker/Dockerfile.CentOS6, yes